pkg/k8sclient/keyed_queue.go

```diff
@@ -20,16 +20,23 @@ import (
 	"sync"
 )
 
+// Queue is an interface which abstracts a queue.
 type Queue interface {
+	// Adds enqueues a key and its associated item.
 	Add(key interface{}, item interface{})
+	// Get removes an item from the queue.
```

Comment:
Looks something odd that `Get` removes an item from the queue. is this true?

Reply:
Hmmm...

```
Get removes an item from the queue and insets the item to the currently processing key set.
```

What about the new comment message?